### PR TITLE
Refactor: (Var|Func)Declaration.checkNestedReference

### PR DESCRIPTION
**Pull request by 9rnsr**
This is the preparation for later clenaup.

### CODE REVIEW

**Comment by 9rnsr** [comment]
You can see the diff view ignoring indent changes.
https://github.com/D-Programming-Language/dmd/pull/5175/files?w=1

File: src/declaration.d [hunk 1]
```diff
@@ -2207,68 +2210,70 @@ public:
...
+
+        /* __require and __ensure will always get called directly,
+         * so they never make outer functions closure.
+         */
+        if (fdthis.ident == Id.require || fdthis.ident == Id.ensure)
+            return false;
+
+        //printf("\tfdv = %s\n", fdv.toChars());
+        //printf("\tfdthis = %s\n", fdthis.toChars());
+        if (loc.filename)
```
Comment (9rnsr) [inline src/declaration.d]: This `loc.filename` check smells hacky. After this refactoring PR is merged, I'll open a cleanup PR for this.

**Comment by ibuclaw** [comment]
OK.

**Comment by ibuclaw** [comment]
Auto-merge toggled on

**Pull request merged**